### PR TITLE
#438 Add option to delete an image

### PR DESCRIPTION
**Pull request by NejcS**
#438

This PR adds a mutation for deleting a crag or route image. The mutation will delete the image entry from the DB and all of the matching images on the disk.

You can test it with the [matching web PR](https://github.com/plezanje-net/web/pull/441).

### CODE REVIEW

File: src/crags/resolvers/crags.resolver.ts [hunk 1]
```diff
@@ -220,4 +222,15 @@ export class CragsResolver {
   ) {
     return this.cragsService.getPopularCrags(dateFrom, top, user != null);
   }
+
+  @Mutation(() => Boolean)
+  @UseGuards(UserAuthGuard)
+  async deleteImage(@CurrentUser() user: User, @Args('id') id: string) {
+    const image = await this.imagesService.findOneById(id);
+    if ((await image.user).id !== user.id) {
```
Comment (salamca) [inline src/crags/resolvers/crags.resolver.ts:230]: Rather throw ForbiddenException ?

Comment (NejcS) [reply src/crags/resolvers/crags.resolver.ts:230]: Good idea

**Comment by NejcS** [comment]
I pushed the change and also added back "graphql" since it was a peer dependency to a couple of other packages and the build was failing.

**Review by demshy — APPROVED** [review]
tested, all good

**Pull request merged**